### ISSUE TITLE
feature: Show headings (#, ##, ###) as different sizes

### ISSUE THREAD

**Issue by tontinton** [MeanderingProgrammer/markdown.nvim#63]
### Is your feature request related to a problem? Please describe.

Today we just see the number of # in the header, but it doesn't let you feel how big it will be relative to the others.

### Describe the solution you'd like

Render the headings text with different font sizes?

### Describe alternatives you've considered

?

### Additional information

Thought it can improve, BTW awesome plugin! I really enjoy it.

**Comment by MeanderingProgrammer** [comment]
I'm glad you're liking it 🙂
Unfortunately that is literally impossible in a terminal. I think theoretically it could be but no terminal emulator I am aware of (Wezterm, Kitty, Alacritty, iTerm, etc) supports multiple font sizes on the same screen. Text is placed in a grid with the size configured. 
It would be a neat, but it is one of the limitations that comes from not being a GUI.
But if you have another way that the level could be indicated let me know 🙂

**Comment by michaelfortunato** [comment]
Would love this feature too yeah. Wondering if neovide could do it. Sounds like a lot needs to be done upstream before we could consider adding this.

**Comment by michaelfortunato** [comment]
Actually it seems like Kitty (of course) at least supports this now. Pretty cool! https://github.com/kovidgoyal/kitty/issues/8226
https://sw.kovidgoyal.net/kitty/text-sizing-protocol/

**Comment by MeanderingProgrammer** [comment]
Yeah, the Kitty feature was added a bit later: https://github.com/MeanderingProgrammer/render-markdown.nvim/issues/366

I will definitely pick this up if support for this is added in neovim core. I don't want to output ansi sequences directly, that would likely cause weird buffer rendering issues, ideally support for this is added in something like highlight groups, and I'll make any needed adjustments. But really depends on how / if it gets adopted.

**Comment by michaelfortunato** [comment]
Totally makes sense. Thanks!
